db/repos/blocks.js

```diff
@@ -274,16 +274,15 @@ class BlocksRepository {
 	}
 
 	/**
-	 * Get multiple blocks to be transported to peers.
+	 * Get block to be transported to peers.
 	 *
-	 * @param {string} ids - Comma separated string of ids
+	 * @param {string} id - id of the block
 	 * @returns {Promise}
 	 * @todo Add description for the return value
 	 *
 	 */
-	getBlocksForTransport(ids) {
-		// TODO: Must use a result-specific method, not .query
-		return this.db.query(sql.getBlocksForTransport, [ids]);
+	getBlockForTransport(id) {
+		return this.db.oneOrNone(sql.getBlockForTransport, [id]);
```

Comment:
Then it doesn't need to be an array. Just pass `id`. I see the sql query already expects a single id (matching by `=` operator instead of `IN`.

Reply:
I am passing it in the form of an array because that's how pgpromise expects parameters to be passed for query; using positional array parameters.. which means that first element of the array will be the first query parameter and so on and so forth.

Reply:
Fair enough :)